docs/best-practices/index.md

```diff
@@ -0,0 +1,438 @@
+
+To mitigate these risks, it's recommended to use secure secrets management tools and practices. Terraform supports various mechanisms for securely managing secrets, including environment variables, encrypted state files, and integration with dedicated secrets management systems like AWS Secrets Manager, Azure Key Vault, or HashiCorp Vault. These tools provide controlled access to secrets, audit trails, and the ability to rotate secrets periodically or in response to a breach.
+
+By keeping secrets out of source control and employing robust secrets management strategies, users can significantly enhance the security posture of their infrastructure deployments. This approach not only protects sensitive information but also aligns with compliance requirements and best practices for secure infrastructure management.
+
+## Multi-team Development
+
+### Use "On-Demand" Development Environments
+
+The recommended approach for multi-team development, when using a GitOps CICD promotion process, is to spin up "on-demand" development and test environments (where possible), specific to new features or to individual teams, to allow for development and integration testing that doesn't conflict with other team's development and test activities.  The Terraform provider allows administrators to use CICD automation to provision new environments as required, and remove them once the project activity no longer needs them.
+
+In a GitOps CICD promotion pipeline, configuration can be translated to Terraform config-as-code and then merged (with Pull Requests) with common test environments, where automated tests can be run.  This then allows the activities in the "on-demand" environments to be merged into a common promotion pipeline to production environments.
+
+In some cases there may be a lack of available integrated systems that cannot be spun up easily or cannot be integrated with.  For example, this may apply to integrated HR systems, or systems that have been installed onto "bare metal" infrastructure.  In these cases, where possible, connected and unrelated systems can be "stubbed" in the spin-up process, and tested during the "Integration test" phase of the project when changes have been merged into a common promotion pipeline.
+
+In some cases it may not be practical to spin up on-demand development or test environments due to impact on project costs, commercial limitations or limitations in the CI/CD processes.  In this case, it is recommended to create static development environments that are ultimately shared between teams/projects and process introduced to mitigate conflicts.  Ideally these development environments (that doesn't impact project work) have their configuration periodically refreshed and aligned with that of common test environments further down the CI/CD promotion pipeline.  Ensure this activity is appropriately scheduled with the project teams involved to avoid wiping configuration that is still in active development.
```

Comment:
In some cases, it is not practical or possible to use on-demand environments.  In these situations, one option is to create static development environments that are shared between teams/projects, along with processes to mitigate conflicts.  Ideally these development environments will have their configuration periodically refreshed and aligned with that of common test environments further down the CI/CD promotion pipeline.  When using a shared environment, ensure the update activity is appropriately scheduled with the project teams involved to avoid wiping configuration that is still in active development.